File: pascal/classes/objects.pas
program Objects;

type MoneyBox = object
  public
    name: string;
    saved_amount: double;
    procedure save(amount: double);
end;

procedure MoneyBox.save(amount: double);
begin
  saved_amount := saved_amount + amount;
end;

var
  piggy: MoneyBox;

begin
  piggy.name := 'Miss Piggy';

  writeln('Money box name: ', piggy.name);
  writeln('Money box saved amount: ', piggy.saved_amount:0:2);

  piggy.save(20);

  writeln('Money box saved amount: ', piggy.saved_amount:0:2);

  piggy.save(100);

  writeln('Money box saved amount: ', piggy.saved_amount:0:2);
end.

File: javascript/classes/objects.js
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

let piggy = {
  name: "",
  saved_amount: 0,
  save: function (amount) {
    this.saved_amount += amount;
  }
}

piggy.name = "Miss Piggy";

console.log("Money box name: " + piggy.name);
console.log("Money box saved amount: " + piggy.saved_amount);

piggy.save(20);

console.log("Money box saved amount: " + piggy.saved_amount);

piggy.save(100);

console.log("Money box saved amount: " + piggy.saved_amount);
